src/api/tasks/tasksApi.ts

```diff
@@ -0,0 +1,160 @@
+    EMail: z.string(),
+  }),
+  Role: z.string(),
+  Status: z.string().optional(),
+  SkippedBy: z
+    .object({
+      Id: z.number().int().positive(),
+      Title: z.string(),
+      EMail: z.string(),
+    })
+    .optional(),
+  Modified: z.coerce.date().optional(),
+});
+type Task = z.infer<typeof Task>;
+
+export const useTasks = (subSite: string, pcolId: string) => {
+  return useQuery({
+    queryKey: ["tasks", subSite, pcolId],
+    queryFn: () =>
+      subWebContext(subSite)
+        .web.lists.getByTitle("tasks")
+        .items.select(
+          "Id",
+          "Title",
+          "pcolId",
+          "Person/Id",
+          "Person/Title",
+          "Person/EMail",
+          "Role",
+          "Status",
+          "SkippedBy/Id",
+          "SkippedBy/Title",
+          "SkippedBy/EMail",
+          "Modified"
+        )
+        .expand("Person", "SkippedBy")<Task[]>(),
+  });
+};
+
+export const useAddTasks = (subSite?: string, pcolId?: string) => {
+  const queryClient = useQueryClient();
+  return useMutation({
+    mutationFn: async (wfDetails: WorkflowDetails) => {
+      if (!subSite) {
+        return Promise.reject("Invalid site");
+      }
+      if (!pcolId) {
+        return Promise.reject("Invalid PCOL");
+      }
+      const [batched, execute] = subWebContext(subSite).batched();
+      const batch = batched.web.lists.getByTitle("tasks");
+
+      for (const reviewer of wfDetails.ParallelReviewers) {
+        if (reviewer.EMail) {
+          const PersonId = await resolvePerson(reviewer);
+          batch.items.add({
+            Title: "",
+            pcolId,
+            PersonId,
+            Role: "Parallel",
+          });
+        }
+      }
+
+      let index = 0;
+      for (const reviewer of wfDetails.SerialReviewers) {
+        if (reviewer.EMail) {
+          const PersonId = await resolvePerson(reviewer);
+          batch.items.add({
+            Title: String(index),
+            pcolId,
+            PersonId,
+            Role: "Serial",
+          });
+          index += 1;
+        }
+      }
+
+      if (wfDetails.OrgReviewer?.EMail) {
+        const PersonId = await resolvePerson(wfDetails.OrgReviewer);
+        batch.items.add({
+          Title: "",
+          pcolId,
+          PersonId,
+          Role: "Org",
+        });
+      }
+
+      if (wfDetails.PCO?.EMail) {
+        const PersonId = await resolvePerson(wfDetails.PCO);
+        batch.items.add({
+          Title: "",
+          pcolId,
+          PersonId,
+          Role: "PCO",
+        });
+      }
+
+      if (wfDetails.Distributor?.EMail) {
+        const PersonId = await resolvePerson(wfDetails.Distributor);
+        batch.items.add({
+          Title: "",
+          pcolId,
+          PersonId,
+          Role: "Distributor",
+        });
+      }
+
+      return execute();
+    },
+    onSuccess: (_data, variables) => {
+      let newStage = "Draft";
+      if (
+        variables.SerialReviewers.length > 0 ||
```

Comment:
So, if they click "Add" for Serial Reviewr, it creates a Serial Reviewer row, and doesn't allow them to delete it, even if they no longer want to do a Serial Review. Not a big deal, because you catch it in the adding tasks, and just ignore it, instead of creating a task not assigned to anyone. Same with if they have blank rows in between filled rows. However, you will also need to catch it when moving to the next phase, because it will see the array, and then set the next phase to "Peer Reivew" even though there are no Peer Reviewers (assuming they also had no Parallel ones)